### DISCUSSION TITLE
How to check if a job is running by its id?

### DISCUSSION THREAD

**Discussion by blastbeng** [agronholm/apscheduler#971] [Q&A]
Hi,
I am developing an app where multiconcurrency is not an option.

I have a scheduled some jobs, which runs on a timer
I have a GUI where an user can manually run the job, but if it's already running I want to alert the user that the job is already running

But I haven't found in the documentation a way to understand if a job is running or not..

Are there any functions I can use to understand if this job is running?

**Comment by legout** [comment]
Hi, 

which version of apscheduler do you use?

Do you mean a long running job, or a schedule for a job?

In v4 you can do the following:

The first one could be checked by looking for the job id in the output of `scheduler.get_jobs()`. For the latter you can use `scheduler.get_schedules()`

```python

from apscheduler import AsyncScheduler
from apscheduler.triggers.interval import IntervalTrigger
from time import sleep

def task():
    sleep(10)

# Check jobs
async with AsyncScheduler() as sched:
     id_ = await sched.add_job(task)
     await sched.start_in_background()
     jobs = await sched.get_jobs()
     print(jobs)

# Check schedules
async with AsyncScheduler() as sched:
    id_ = await sched.add_schedule(task, IntervalTrigger(seconds=30))
    await sched.start_in_background()
    scheds = await sched.get_schedules()
    print(scheds)
```

**Comment by Jzhenli** [comment]
@legout I‘m using the V3.11.0 version, if run a long time async task, how to check if the job is running or not？

**Reply by HomerusJa** [reply]
As far as I know this is a feature which will be added in v4.0.0, and I currently don’t know how you would do this in v3.x. You might be able to do that with a custom executor, but this would definitely be a hack and I’d suggest to wait for v4.0.0 if possible. @agronholm Might there be a way you know of how to do this?

**Reply by Jzhenli** [reply]
@HomerusJa  thanks for your reply, seems v4 is not ready to release  and v3 is the latest stable version now.

**Reply by HomerusJa** [reply]
Yea, that’s correct

**Reply by HomerusJa** [reply]
> **Limiting the number of concurrently executing instances of a job**
By default, only one instance of each job is allowed to be run at the same time. This means that if the job is about to be run but the previous run hasn’t finished yet, then the latest run is considered a misfire. It is possible to set the maximum number of instances for a particular job that the scheduler will let run concurrently, by using the max_instances keyword argument when adding the job.

Maybe that behavior also works for you. You could now listen to the EVENT_JOB_MAX_INSTANCES and alert the user that his job has already begun.
